pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -202,9 +199,14 @@ func (m *Manager) aquireInhibitLock() error {
 	return nil
 }
 
+// Returns if the feature is enabled
+func (m *Manager) isFeatureEnabled() bool {
+	return utilfeature.DefaultFeatureGate.Enabled(features.GracefulNodeShutdown) && m.shutdownGracePeriodRequested > 0
```

Comment:
Maybe have to judge `m.shutdownGracePeriodCriticalPods > 0 && m.shutdownGracePeriodCriticalPods < m.shutdownGracePeriodRequested`?

Reply:
This should already be covered in the kubelet config validation I think:

https://github.com/kubernetes/kubernetes/blob/102124464a994946e151c976775cf751423b14f7/pkg/kubelet/apis/config/validation/validation.go#L143-L156

Reply:
It looks good, it is best to add a verification of `kc.ShutdownGracePeriodCriticalPods.Duration >= kc.ShutdownGracePeriodRequested.Duration`

Reply:
I might not fully understand your comment, we already have a check that if `kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration` an error is returned in the validation.  

As a result, I'm not clear why it would make sense to add inside `isFeatureEnabled` since if someone specifies `CriticalPods.Duration` > `ShutdownGracePeriod` kubelet config validation will fail, so it kubelet won't startup (and thus this code won't even run).

Reply:
Oh I see. looks great!